File: python/链表/1474. 删除链表 M 个节点之后的 N 个节点.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def deleteNodes(self, head: ListNode, m: int, n: int) -> ListNode:
        start = ListNode(-1)
        start.next = head
        pre,cur = start,start.next
        while cur:
            i = 0
            while cur and i != m:
                cur = cur.next
                pre = pre.next
                i+=1
            i = 0
            while cur and i != n:
                cur = cur.next
                i+=1
            pre.next = cur
        return head

File: go/link/1474. 删除链表 M 个节点之后的 N 个节点.go
package link

func deleteNodes(head *ListNode, m int, n int) *ListNode {
	start := &ListNode{Val: -1}
	start.Next = head
	pre, cur := start, start.Next
	for cur != nil {
		i := 0
		for cur != nil && i != m {
			cur = cur.Next
			pre = pre.Next
			i++
		}
		i = 0
		for cur != nil && i != n {
			cur = cur.Next
			i++
		}
		pre.Next = cur
	}
	return head
}
